monitor/src/bin/queue.rs

```diff
@@ -0,0 +1,615 @@
+    let config = TOML
+        .queue
+        .as_ref()
+        .ok_or_eyre("monitor.toml has no [queue]!")?;
+    let mut queue = Queue::default();
+
+    loop {
+        info!("Querying servers for updates");
+        queue.start_update();
+
+        let mut set = JoinSet::new();
+        for server_url in config.servers.iter() {
+            set.spawn(async {
+                (
+                    Server(server_url.clone()),
+                    get_monitor_dashboard_for_server(server_url).await,
+                )
+            });
+        }
+
+        for (server, result) in set.join_all().await {
+            match result {
+                Ok(response) => {
+                    debug!(?server, ?response);
+                    queue.servers.insert(
+                        server,
+                        ServerStatus {
+                            last_monitor_response: response,
+                            fresh: true,
+                        },
+                    );
+                }
+                Err(error) => {
+                    error!(?error);
+                }
+            }
+        }
+
+        let mut queue_text = String::default();
+        for (unique_id, entry) in queue.iter() {
+            let access_times = ACCESS_TIMES.read().expect("Poisoned");
+            let access_time = access_times.get(unique_id).expect("Guaranteed by Queue");
+            writeln!(
+                &mut queue_text,
+                "- {unique_id} (last request {:?} ago)",
+                access_time.elapsed()
+            )?;
+            writeln!(&mut queue_text, "  {entry:?}")?;
+        }
+        *QUEUE_CACHE.write().expect("Poisoned") = queue
+            .iter()
+            .flat_map(|(unique_id, entry)| {
+                queue
+                    .quick_lookup_info(entry)
+                    .map(|info| (unique_id.clone(), info))
+            })
+            .collect();
+
+        let mut servers_text = String::default();
+        for (server, status) in queue.servers.iter() {
+            write!(&mut servers_text, "- {server}")?;
+            if status.fresh {
+                writeln!(&mut servers_text, "")?;
+            } else {
+                writeln!(&mut servers_text, " (stale!)")?;
+            }
+            for (profile_key, runner_counts) in status.fresh_or_stale().profile_runner_counts.iter()
+            {
+                writeln!(&mut servers_text, "    - {profile_key}")?;
+                writeln!(
+                    &mut servers_text,
+                    "      {} idle, {} healthy, {} target",
+                    runner_counts.idle, runner_counts.healthy, runner_counts.target
+                )?;
+            }
+        }
+
+        let mut new_dashboard = String::default();
+        writeln!(&mut new_dashboard, ">>> queue\n{queue_text}")?;
+        writeln!(&mut new_dashboard, ">>> servers\n{servers_text}")?;
+        *DASHBOARD.write().expect("Poisoned") = Some(new_dashboard);
```

Comment:
comment(non-blocking): Not sure if we really benefit doing this here, as we could just do this on request and cache the result for some time to prevent running this too many times.